shared/hooks/router/use-update-search-params.ts

```diff
@@ -0,0 +1,17 @@
+import { usePathname, useRouter, useSearchParams } from "next/navigation";
+import { useCallback } from "react";
+
+export const useUpdateSearchParams = () => {
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+  const router = useRouter();
+  return useCallback(
```

Comment:
Could return an object with `updateSearchParams` method and a `getSearchParams` value so we don't have to import `useSearchParams` when we use this hook.